### PR TITLE
Updating the Zenodo handling

### LINKED ISSUE #161: Cleaner error handling for `create_deposit` entry point
At the moment if a record ID is supplied to instead of a concept ID to the `create_deposit` entry point, it fails in a very hard to read way. We should improve the error handling so that the user can easily understand what is going wrong

### PR DESCRIPTION
**Pull request by davidorme**
This PR expands a bit of the actual issue described in #161 after feedback from Sam.

* Sam's error came from using outdated/incorrect URLs for the Zenodo API endpoints. I've done two things:
    * Removed the Zenodo API endpoints from the configuration file and hard-coded them. Seemed oddly fussy and not something that changes (often!) and this avoids a simple configuration issue.
    * Updated  the Zenodo error handling to report some more basic information and then adding additional JSON data only if available (not all zenodo responses provide additional JSON info),
* I've added mutually exclusive `--live` and `--sandbox` switches to the Zenodo commands to allow data managers to switch the target publication site without having to maintain different config files.
* I've update the "Using safedata" section for Data Managers to include separate sections on validation, publication and posting metadata, with more discussion of the key points and more example outputs.
* Updated the code to handle the change in the new version generation on Zenodo - this has shifted from creating an empty deposit from the concept ID to creating a clone from the most recent record ID. The `publish_dataset` function now does some filename and hash checking to work out how to align the files being provided with what is already online. Docs updated to reflect this.

Closes #161

### CODE REVIEW

**Comment by davidorme** [comment]
@jacobcook1995 I haven't tested this stuff, but it does it look sensible? Still need to figure out what went wrong with Sam.

File: safedata_validator/zenodo.py [hunk 1]
```diff
@@ -182,16 +195,17 @@ def get_deposit(
...
     """Create a new deposit.
 
     Creates a new deposit draft, possibly as a new version of an existing published
     record.
 
     Args:
-        concept_id: An optional concept id of a published record to create a new version
-            of an existing dataset.
+        new_version: Optionally, create a new version of the dataset with the provided
+            Zenodo ID. This must be the most recent version of the dataset concept.
```
Comment (jacobcook1995) [inline safedata_validator/zenodo.py:208]: What does "most recent version of the dataset concept" mean?

**Review by jacobcook1995 — APPROVED** [review]
Yes, looks sensible to me. Had one minor comment

File: docs/data_managers/using_safedata/publishing_datasets.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+command](../command_line_tools/safedata_zenodo.md#the-publish_dataset-subcommand) is
+the main function for publishing a dataset. The example below shows it being used to
+publish a dataset and additional external files:
+
+```sh
+safedata_zenodo publish_dataset Example.xlsx -e  Supplementary_files.zip
+```
+
+That command packages up a set of operations needed to publish a dataset. The tabs
+examples below show the underlying workflow, either using the command line interface or
```
Comment (jacobcook1995) [inline docs/data_managers/using_safedata/publishing_datasets.md:28]: "tabs examples"

File: docs/data_managers/using_safedata/publishing_datasets.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+Zenodo can hold multiple versions of a dataset, allowing you to publish updates and
+corrections. When you create a new dataset, that dataset is assigned a record id but
+is also automatically also given a **concept record id**. These are often sequential:
+in the previous example, the specific record ID for the dataset was `1143714` - the
+concept record ID might be `1143713`. The concept record id is used to group versions of
+a dataset and the associated DOI is used to redirect to the most recent version of the
+dataset.
+
+Releasing a new version generates a copy of the most recent version of a dataset,
```
Comment (jacobcook1995) [inline docs/data_managers/using_safedata/publishing_datasets.md:60]: When you say "generates a copy" do you mean locally or does it upload it to Zenodo?

File: docs/data_managers/using_safedata/posting_metadata.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Updating the metadata server
+
+## Posting data for new published datasets
```
Comment (jacobcook1995) [inline docs/data_managers/using_safedata/posting_metadata.md:3]: Would be good to include a "what if I don't have a metadata server but might have one in the future?" section somewhere

**Review by jacobcook1995 — COMMENTED** [review]
Looks like a big improvement, just had a few comments

File: docs/data_managers/using_safedata/publishing_datasets.md (hunk 3 above)
Comment (davidorme) [reply docs/data_managers/using_safedata/publishing_datasets.md:60]: It's created on Zenodo. Clarified

**Comment by davidorme** [comment]
@jacobcook1995 This is changed again with the update to match the changed Zenodo new version process. I think this is a sensible resolution (use `md5` and filename to check for new/changed/unchanged/removed files) and I've updated the docs to match.

**Review by jacobcook1995 — APPROVED** [review]
I've look through the changes and they all look good to me!

**Pull request merged**